Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryISISLoadAndProcess.py

```diff
@@ -0,0 +1,450 @@
+        self.log().information('Slicing workspace ' + workspace + ' into ' + sliced_workspace_name)
+        self._runSliceAlgorithm(workspace, sliced_workspace_name)
+        return sliced_workspace_name
+
+    def _getSlicedWorkspaceGroupName(self, workspace):
+        return workspace + '_sliced'
+
+    def _setChildAlgorithmPropertyIfProvided(self, alg, property_name):
+        """Set the given property on the given algorithm if it is set in our
+        inputs. Leave it unset otherwise."""
+        if not self.getProperty(property_name).isDefault:
+            alg.setProperty(property_name, self.getPropertyValue(property_name))
+
+    def _reduce(self, input_workspace, first_trans_workspace, second_trans_workspace):
+        """Run the child algorithm to do the reduction. Return the child algorithm."""
+        self.log().information('Running ReflectometryReductionOneAuto on ' + input_workspace)
+        alg = self.createChildAlgorithm("ReflectometryReductionOneAuto")
+        for property in self._reduction_properties:
+            alg.setProperty(property, self.getPropertyValue(property))
+        self._setChildAlgorithmPropertyIfProvided(alg, Prop.OUTPUT_WS)
+        self._setChildAlgorithmPropertyIfProvided(alg, Prop.OUTPUT_WS_BINNED)
+        self._setChildAlgorithmPropertyIfProvided(alg, Prop.OUTPUT_WS_LAM)
+        alg.setProperty("InputWorkspace", input_workspace)
+        alg.setProperty("FirstTransmissionRun", first_trans_workspace)
+        alg.setProperty("SecondTransmissionRun", second_trans_workspace)
+        alg.execute()
+        return alg
+
+    def _removePrefix(self, workspace, isTrans):
+        """Remove the TOF prefix from the given workspace name"""
+        prefix = self._transPrefix if isTrans else self._tofPrefix
+        prefix_len = len(prefix)
+        name_start = workspace[:prefix_len]
+        if len(workspace) > prefix_len and name_start == prefix:
+            return workspace[prefix_len:]
+        else:
+            return workspace
+
+    def _finalize(self, child_alg):
+        """Set our output properties from the results in the given child algorithm"""
+        self._setOutputWorkspace(Prop.OUTPUT_WS, child_alg)
+        self._setOutputWorkspace(Prop.OUTPUT_WS_BINNED, child_alg)
+        self._setOutputWorkspace(Prop.OUTPUT_WS_LAM, child_alg)
+
+    def _setOutputWorkspace(self, property_name, child_alg):
+        """Set the given output property from the result in the given child algorithm,
+        if it exists"""
+        value = child_alg.getPropertyValue(property_name)
+        if value:
+            self.setPropertyValue(property_name, value)
+            self.setProperty(property_name, child_alg.getProperty(property_name).value)
+
+
+def _setRunNumberForWorkspace(workspace_name, run_number):
+    """Set the run number in the sample log for the given workspace"""
+    AddSampleLog(Workspace=workspace_name, LogName='run_number', LogText=str(run_number),
```

Comment:
Same comment as above about `AddSampleLog`. Maybe does not need to be a function then.